From 756f68a9aa1fb708ed175a680bae2a474052b0d8 Mon Sep 17 00:00:00 2001
From: "GONG, Ruiqi" <gongruiqi1@huawei.com>
Date: Tue, 15 Nov 2022 15:36:53 +0800
Subject: [PATCH 0028/1398] [Huawei] kabi: use CONFIG_KABI_RESERVE to control
 reservation in ucount_type and user_table

Offering: HULK
hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4MYYH

----------------------------------------

Commit 328e64817efe ("kabi: reserve space for cred and user_namespace"),
besides reserving space for struct cred and struct user_namespace, adds
new members to enum ucount_type and struct ctl_table user_table[] as
well. Use CONFIG_KABI_RESERVE to control them.

Signed-off-by: GONG, Ruiqi <gongruiqi1@huawei.com>
---
 include/linux/user_namespace.h | 2 ++
 kernel/ucount.c                | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/include/linux/user_namespace.h b/include/linux/user_namespace.h
index f79abb16a874..6bdd18062ee0 100644
--- a/include/linux/user_namespace.h
+++ b/include/linux/user_namespace.h
@@ -51,6 +51,7 @@ enum ucount_type {
 	UCOUNT_INOTIFY_INSTANCES,
 	UCOUNT_INOTIFY_WATCHES,
 #endif
+#ifdef CONFIG_KABI_RESERVE
 	/* These 15 members are reserved (with extra margin) for the future
 	 * enlargement of enum ucount_type, as how RH8.1 did it. This number
 	 * should be enough, as 6 of them are very likely to be used in the near
@@ -71,6 +72,7 @@ enum ucount_type {
 	UCOUNT_KABI_RESERVE13,
 	UCOUNT_KABI_RESERVE14,
 	UCOUNT_KABI_RESERVE15,
+#endif
 	UCOUNT_COUNTS,
 };
 
diff --git a/kernel/ucount.c b/kernel/ucount.c
index 1f5825b674d8..093b2fa76385 100644
--- a/kernel/ucount.c
+++ b/kernel/ucount.c
@@ -77,6 +77,7 @@ static struct ctl_table user_table[] = {
 	UCOUNT_ENTRY("max_inotify_instances"),
 	UCOUNT_ENTRY("max_inotify_watches"),
 #endif
+#ifdef CONFIG_KABI_RESERVE
 	/* These corresponds to the reservation in enum ucount_type */
 	{ }, // UCOUNT_KABI_RESERVE1
 	{ }, // UCOUNT_KABI_RESERVE2
@@ -93,6 +94,7 @@ static struct ctl_table user_table[] = {
 	{ }, // UCOUNT_KABI_RESERVE13
 	{ }, // UCOUNT_KABI_RESERVE14
 	{ }, // UCOUNT_KABI_RESERVE15
+#endif
 	{ }
 };
 #endif /* CONFIG_SYSCTL */
-- 
2.17.1

